packages/core/src/actions/actions.ts

```diff
@@ -56,6 +56,66 @@ export const UPDATE_I18N = 'jsonforms/UPDATE_I18N' as const;
 export const ADD_DEFAULT_DATA = 'jsonforms/ADD_DEFAULT_DATA' as const;
 export const REMOVE_DEFAULT_DATA = 'jsonforms/REMOVE_DEFAULT_DATA' as const;
 
+export type UpdateArrayContext =
+  | { type: 'ADD'; values: any[] }
```

Comment:
Instead of only specifying the values, can we also specify the array index at which they will be added? We already have consumers who customized their array add operations to for example add them at the beginning of the array. That way we could cover them too.

The remove and move information is sufficient :+1:

Reply:
I don't think we can, as we don't have that information in the dispatch method. See: https://github.com/eclipsesource/jsonforms/pull/2317/files#diff-a1bd17d367230eda9ec6010e4de56595495fd67e51d4faa5650ec2d800b9e702R836

Am I missing something?

Reply:
Ah yes, you are right, I missed that!